apps/src/code-studio/assets/SpriteUpload.jsx

```diff
@@ -1,330 +1,520 @@
 import React from 'react';
 import color from '@cdo/apps/util/color';
 import {makeEnum} from '@cdo/apps/utils';
+import ValidationStep, {Status} from '../../lib/ui/ValidationStep';
 import {
   getManifest,
   getLevelAnimationsFiles,
-  uploadSpriteToAnimationLibrary,
+  uploadAnimationToAnimationLibrary,
   uploadMetadataToAnimationLibrary
 } from '@cdo/apps/assetManagement/animationLibraryApi';
 
-const SpriteLocation = makeEnum('library', 'level');
-const UploadStatus = makeEnum(
-  'fail',
-  'success',
-  'filenameOverride',
-  'badFilename',
-  'none'
-);
+const EXTENSION_CHECK = 'extensionError';
+const FILENAME_CHECK = 'filenameError';
+const FILESIZE_CHECK = 'filesizeError';
+const ANIMATION_EXISTS_CHECK = 'animationExistsError';
+const AnimationLocations = makeEnum('library', 'level');
 
-// Levelbuilder tool for adding sprites to the Spritelab animation library.
-export default class SpriteUpload extends React.Component {
+const initialSpriteLabLibrariesState = {
+  spriteLabLevelAnimations: [],
+  spriteLabLibraryCategories: [],
+  spriteLabLibraryAnimations: []
+};
+
+const newImageState = {
+  fileData: null,
+  filename: '',
+  filePreviewURL: ''
+};
+
+const imageChecksInitialState = {
+  [EXTENSION_CHECK]: Status.WAITING,
+  [FILENAME_CHECK]: Status.WAITING,
+  [FILESIZE_CHECK]: Status.WAITING,
+  [ANIMATION_EXISTS_CHECK]: Status.WAITING
+};
+
+const initialUserInputMetaData = {
+  availability: '',
+  category: '',
+  aliases: []
+};
+
+const initialGeneratedMetadata = {
```

Comment:
I like this chunking of state. I know it's an intermediate phase until we start to break this file out into smaller components, but I think this is a good organization strategy for right now.